src/compiler/checker.ts

```diff
@@ -11336,6 +11336,11 @@ namespace ts {
         function getTypeFromTypeReference(node: TypeReferenceType): Type {
             const links = getNodeLinks(node);
             if (!links.resolvedType) {
+                // handle LS queries on the `const` in `x as const` by resolving to the type of `x`

```

Comment:
Instead of this shouldn't `getTypeOfNode` just check if it is `const` of ` as const` expression should forward it to correct node instead?

Reply:
Ehhhhhh, it _could_, but I'd rather push the handling to here, to prevent us from ever attempting to resolve the `const` in `as const` as a real type reference, however we may get there in the future.